src/clr-angular/data/datagrid/datagrid-row.ts

```diff
@@ -168,6 +194,14 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit
   }
 
   ngAfterViewInit() {
+    this.subscriptions.push(
+      this.expand.animate.subscribe(() => {
+        this.activeHeight = 'auto';
+        setTimeout(() => {
```

Comment:
`setTimeout` is no go for us as this will trigger change detection through the whole application. I am also wondering why you needed this.

Reply:
As we discussed offline, there is certain lifecycle issue similar to those we observed in other cases (like the tree) when used in conjunction with the clrIfExpanded directive.
I will add a new bug for tracking a possible improvement research, once this may get pushed.
The possible deprecation scenario of the expanded rows will take some time, so as we already have this nearly done, I'd prefer to push it and remove the hack. If someone objects this, please, post a clear no-go for this whole PR, so I can close it and move forward.

Reply:
I would still press on this to get removed. I think if not the main, it's one of the main improvements we can get from this PR is removal of this setTimeout. Having this setTimeout in this PR, we are still dragging some part of the old hack to this new animation solution.

As a library, Clarity shouldn't be triggering extra change detections throughout the entirety of a user's application.
Is there any way we could time the animation without triggering change detection through the entire app?

Reply:
The last suggested rework removes the need for setTimeout or detectChanges. It introduces a new intermediary component that will handle the animations. It extracts the complexities of the content-based animations and can be reused.